### PR TITLE
Supporting PHP 8.1 enums

### PR DESCRIPTION
**Pull request by dkreuer**
This PR provides support for PHP 8.1 enums.

With composer-require-checker up to 3.6.0 enum definitions lead to a parsing error and therefore unknown symbols.

Taken over from #320 with same implementation but adding tests.

### CODE REVIEW

**Review by DanielBadura — APPROVED** [review]

File: test/ComposerRequireCheckerTest/Cli/CheckCommandTest.php [hunk 1]
```diff
@@ -207,4 +207,21 @@ public function testNoUnknownComposerSymbolFound(): void
             $this->commandTester->getDisplay()
         );
     }
+
+    public function testNoUnknownEnumSymbolsFound(): void
+    {
+        if (version_compare(PHP_VERSION, '8.1.0') === -1) {
```
Comment (Ocramius) [inline test/ComposerRequireCheckerTest/Cli/CheckCommandTest.php]: This should be `@requires php >= 8.1.0` (https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api)

File: src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php [hunk 2]
```diff
@@ -60,6 +61,15 @@ private function recordClassDefinition(Node $node): void
         $this->recordDefinitionOf($node);
     }
 
+    private function recordEnumDefinition(Node $node): void
```
Comment (Ocramius) [inline src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php:64]: Is a unit test possible here? :thinking:

**Review by Ocramius — CHANGES_REQUESTED** [review]
Thanks for writing an integration test for this! Some minor changes requested, but otherwise good to go.

File: test/ComposerRequireCheckerTest/Cli/CheckCommandTest.php (hunk 1 above)
Comment (dkreuer) [reply test/ComposerRequireCheckerTest/Cli/CheckCommandTest.php]: Switched condition to annotation.

File: src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php (hunk 2 above)
Comment (dkreuer) [reply src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php:64]: Added a unit test in `test/ComposerRequireCheckerTest/NodeVisitor/DefinedSymbolCollectorTest.php`

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @dkreuer!

**Comment by Ocramius** [comment]
🚢

**Pull request merged**